src/types/InputTypes.ts

```diff
@@ -1,6 +1,10 @@
+import { ChangeEvent } from 'react';
 import { UIPropsType } from './CommonTypes';
 
-export interface InputType extends UIPropsType {
+export interface InputType extends Partial<UIPropsType> {
+  value: string;
   type: 'email' | 'text' | 'password';
+  handleOnChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  placeholder: string;
```

Comment:
I would make a placeholder as an optional prop because sometimes we will use labels